### ISSUE TITLE
E0 Not functioning

### ISSUE THREAD

**Issue by FullTiltt** [bigtreetech/BIGTREETECH-OCTOPUS-V1.0#88]
The issue I've had was after swapping in the octopus board on a previously working system was that the extruder did not extrude when it was plugged into e0 
#define E0_DRIVER_TYPE TMC2209
was set and plugged into board in the correct pins on the board right jumpers selected etc. In the display ui under tmc settings for voltage or mode i could see it as available and could select it to extrude under move. But obviously it did nothing. If i uncommented the E1 as above and commented out E0 the options stated above disappeared and i was left with only X,Z,Y. Basically long story short is that the solution ended up being that i had to have E0 uncommented as above but it has to be plugged into the pins for motor 4(I have a X, Y, (Z1, Z2 these being split among a single driver as i understand) and a single extruder motor) and it shows up as E1 i guess correctly but has to be set as e0? Can anyone make sense of this?

**Comment by radek8** [comment]
By pin file:
X  -axis - Motor 0
Y  -axis - Motor 1
Z  -axis - Motor 2
Z2-axis - Motor 3
E0-axis - Motor 4

**Comment by radek8** [comment]
Are you involved like this?

![image](https://user-images.githubusercontent.com/46979052/144452393-dc98f35f-d35a-498a-a05b-344cc2a3c138.png)

**Comment by FullTiltt** [comment]
Sorry for late reply, the issue has been resolved by accident. Also after reading through 3dwork.io 's article on the board i saw mention of it there, this being the fact that as well as 2_1- 2_2  motor3 is reserved for second driver for z axis's. I see no mention of it in the documentation, seems like it'd be a good idea for it to be referenced there, as it overly obvious although after seeing the reasoning it does make good sense..

**Comment by radek8** [comment]
Wasn't that clear from my answers?

**Comment by radek8** [comment]
Close the ticket

**Comment by RadicalxEdward** [comment]
This solved my issue trying to get a BTT Octopus V1.1 to work with Marlin on an Ender 3. Couldn't get the extruder to move but it turned out I had to have it configured as:
Driver 0 / Motor 0 = X0
Driver 1 / Motor 1 = Y0
Driver 2 / Motor 2_1 = Z0
Driver 2 / Motor 2_2 = [N/A]
Driver 3 / Motor 3 = [N/A]
Driver 4 / Motor 4 = E0

Which makes absolutely no sense to me. I'd think it would be drivers 0-3 as X, Y, Z, E with driver 2 just driving 2 outputs, so there'd be 1 empty motor output if you didn't have dual z axis motors. 

I did see in the configuration somewhere that there's an option for dual Z *drivers*, but I had that disabled, so i don't understand why it would still reserve driver 3 / motor 3 for another Z axis. And i didn't find any other documentation anywhere that says that's the case including the BTT Octopus manual, install guides, etc. Really should be documented in the manual.
